frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/FileDownloader.java

```diff
@@ -83,7 +88,28 @@ private CloseableHttpResponse execute(String requestUrl) throws IOException {
             return executeViaProxy(proxy, requestUrl);
         } else {
             LOGGER.info("No proxy was configured, downloading directly");
-            response = buildHttpClient(null).execute(new HttpGet(requestUrl));
+            if (userName != null && !"".equals(userName) && password != null && !"".equals(password)) {
```

Comment:
We have `StringUtils.isNotEmpty` from `org.codehaus.plexus.util` that can be used here.